src/pygeoprocessing/geoprocessing.py

```diff
@@ -1034,47 +1034,97 @@ def create_raster_from_vector_extents(
+        raster_driver_creation_tuple=DEFAULT_GTIFF_CREATION_TUPLE_OPTIONS):
+    """Create a raster from a given bounding box.
+
+    Args:
+        target_bounding_box (tuple): a 4-element iterable of (minx, miny,
+            maxx, maxy) in projected units matching the SRS of
+            ``target_srs_wkt``.
+        target_raster_path (string): The path to where the new raster should be
+            created on disk.
+        target_pixel_size (tuple): A 2-element tuple of the (x, y) pixel size
+            of the target raster.  Elements are in units of the target SRS.
+        target_pixel_type (int): The GDAL GDT_* type of the target raster.
+        target_srs_wkt (string): The SRS of the target raster, in Well-Known
+            Text format.
+        target_nodata (float): The nodata value of the target raster, or
+            ``None`` if no nodata value is to be set.
+        fill_value=None (number): If provided, the value that the target raster
+            should be filled with.
+        raster_driver_creation_tuple (tuple): a tuple containing a GDAL driver
+            name string as the first element and a GDAL creation options
+            tuple/list as the second. Defaults to a GTiff driver tuple
+            defined at geoprocessing.DEFAULT_GTIFF_CREATION_TUPLE_OPTIONS.
+
+    Returns:
+        ``None``
+    """
+    if target_pixel_type not in _VALID_GDAL_TYPES:
+        raise ValueError(
+            f'Invalid target type, should be a gdal.GDT_* type, received '
+            f'"{target_pixel_type}"')
+
+    bbox_minx, bbox_miny, bbox_maxx, bbox_maxy = target_bounding_box
+
+    driver = gdal.GetDriverByName(raster_driver_creation_tuple[0])
+    n_bands = 1
     n_cols = int(numpy.ceil(
-        abs((shp_extent[1] - shp_extent[0]) / target_pixel_size[0])))
+        abs((bbox_maxx - bbox_minx) / target_pixel_size[0])))
     n_cols = max(1, n_cols)
 
     n_rows = int(numpy.ceil(
-        abs((shp_extent[3] - shp_extent[2]) / target_pixel_size[1])))
+        abs((bbox_maxy - bbox_miny) / target_pixel_size[1])))
     n_rows = max(1, n_rows)
 
-    driver = gdal.GetDriverByName(raster_driver_creation_tuple[0])
-    n_bands = 1
     raster = driver.Create(
         target_raster_path, n_cols, n_rows, n_bands, target_pixel_type,
         options=raster_driver_creation_tuple[1])
-    raster.GetRasterBand(1).SetNoDataValue(target_nodata)
+    raster.SetProjection(target_srs_wkt)
 
     # Set the transform based on the upper left corner and given pixel
-    # dimensions
+    # dimensions.  Bounding box is in format [minx, miny, maxx, maxy]
     if target_pixel_size[0] < 0:
-        x_source = shp_extent[1]
+        x_source = bbox_maxx
     else:
-        x_source = shp_extent[0]
+        x_source = bbox_minx
     if target_pixel_size[1] < 0:
-        y_source = shp_extent[3]
+        y_source = bbox_maxy
     else:
-        y_source = shp_extent[2]
+        y_source = bbox_miny
```

Comment:
These lines look like good candidates for the ternary operator, if you'd like to condense them

Reply:
Great idea!  Patched in 0ee2720